prometheus_client/exposition.py

```diff
@@ -271,8 +271,20 @@ def write_to_textfile(path, registry):
     tmppath = '%s.%s.%s' % (path, os.getpid(), threading.current_thread().ident)
     with open(tmppath, 'wb') as f:
         f.write(generate_latest(registry))
-    # rename(2) is atomic.
-    os.rename(tmppath, path)
+    
+    # rename(2) is atomic but fails on Windows if the destination file exists
+    if os.name == 'posix':
```

Comment:
I wonder if it is worth keeping the `posix` behavior as the `else` case, and only do the remove/rename or replace logic if os.name is `nt`. That way it is clear that the special behavior is for windows?